main.java

```diff
@@ -0,0 +1,73 @@
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+
+public class FileManager {
+    public static void main(String[] args) {
+        String fileName = "test.txt";
+        String newFileName = "new_test.txt";
+        String data = "Hello, Bito!";
+
+        // Create file
+        try (FileWriter writer = new FileWriter(fileName)) {
+            writer.write(data);
+            writer.flush(); // Bug 1: Unnecessary flush call
+        } catch (IOException e) {
+            System.out.println("Error creating file: " + e.getMessage());
+            return;
+        }
+        System.out.println("File created successfully");
+
+        // Read file
+        try {
+            String content = new String(Files.readAllBytes(Paths.get(fileName)));
+            System.out.println("File read successfully: " + content);
+        } catch (IOException e) {
+            System.out.println("Error reading file: " + e.getMessage());
+            return;
+        }
+
+        // Append to file
+        try (FileWriter writer = new FileWriter(fileName)) { // Bug 2: File opened in overwrite mode instead of append mode
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/rohankokkulabito/cra/pull/2#issuecomment-2200045171>#edae76</a><span> - 07/01/2024, 12:44 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The file is opened in overwrite mode instead of append mode. This will overwrite the existing content of the file instead of appending to it.
</div>
<div id="fix">
<b>Fix:</b> Open the file in append mode by using the 'FileWriter' constructor with the 'append' parameter set to 'true'.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -34,7 +34,7 @@
         // Append to file
 -       try (FileWriter writer = new FileWriter(fileName)) { // Bug 2: File opened in overwrite mode instead of append mode
 +       try (FileWriter writer = new FileWriter(fileName, true)) { // Bug 2: File opened in overwrite mode instead of append mode
             writer.write("\nAppended text.");
         } catch (IOException e) {
             System.out.println("Error appending to file: " + e.getMessage());
```
</div>
</details>
</div>
<br>


</div>